RecoEgamma/EgammaElectronProducers/plugins/TrackingRegionsFromSuperClustersProducer.h

```diff
@@ -247,15 +250,11 @@ getVtxPos(const edm::Event& iEvent,double& deltaZVertex)const
 {
   if(useZInVertex_){
     auto verticesHandle = getHandle(iEvent,verticesToken_);   
-    if(!verticesHandle->empty()){
-      deltaZVertex = originHalfLength_;
-      const auto& pv = verticesHandle->front();
-      return GlobalPoint(pv.x(),pv.y(),pv.z());
-    }
+    deltaZVertex = originHalfLength_;
+    const auto& pv = verticesHandle->front();
+    return GlobalPoint(pv.x(),pv.y(),pv.z());
```

Comment:
you're right, this was a mistake on my part. I was thinking that the collection wasnt there it should throw. But if there is no recoed vertex, it should degrade gracefully.